src/Propel/Generator/Behavior/Sluggable/SluggableBehavior.php

```diff
@@ -291,7 +291,7 @@ protected function makeSlugUnique(\$slug, \$separator = '" . $this->getParameter
             $script .= "
 
         \$count = " . $this->builder->getStubQueryBuilder()->getClassname() . "::create()
-            ->filterBySlug(\$this->$getter())
+            ->filterBySlug(\$slug)
```

Comment:
I dont know, why we dont use the $slug parameter here, but there are even more conditions in this method which need to be adopted in case you are right.